### PR TITLE
Add openfermion conversion module to qml.ops

### PR DESCRIPTION
**Pull request by soranjh**
**Context:**
This PR unifies the functions needed for converting openfermion objects to pennylane observables, also see [here](https://github.com/PennyLaneAI/pennylane/pull/2199).

**Description of the Change:**
The `qml.hf.convert` module is moved inside `qml.ops` and the `qml.qchem` functions are modified to adapt with this change.

**Benefits:**
Helps with decreasing external dependencies.

**Possible Drawbacks:**
NA

**Related GitHub Issues:**
NA

### CODE REVIEW

File: qchem/tests/test_convert.py [hunk 1]
```diff
@@ -18,14 +18,14 @@
...
-
-pytest.importorskip("openfermion")
+from openfermion import QubitOperator
 
 import pennylane as qml
-from openfermion import QubitOperator
 from pennylane import numpy as np
 from pennylane import qchem
 
+pytest.importorskip("openfermion")
```
Comment (ixfoduap) [inline qchem/tests/test_convert.py:27]: Does the order matter for the lines 
`from openfermion import QubitOperator` and `pytest.importorskip("openfermion")`?

**Review by ixfoduap — APPROVED** [review]

**Review by obliviateandsurrender — CHANGES_REQUESTED** [review]

**Comment by obliviateandsurrender** [comment]
Looks great @soranjh. Just left a couple of minor comments.

File: qchem/tests/test_convert.py (hunk 1 above)
Comment (soranjh) [reply qchem/tests/test_convert.py]: Modified it to `openfermion = pytest.importorskip("openfermion")`.

File: qchem/pennylane_qchem/qchem/convert.py [hunk 2]
```diff
@@ -281,7 +281,7 @@ def _openfermion_pennylane_equivalent(
     return openfermion_qubit_operator == _pennylane_to_openfermion(coeffs, ops, wires=wires)
 
 
-def import_operator(qubit_observable, format, wires=None, tol=1e08):
+def import_operator(qubit_observable, format="openfermion", wires=None, tol=1e08):
```
[the remark this answers is not in the document]
Comment (soranjh) [reply qchem/pennylane_qchem/qchem/convert.py:284]: Yes, added a sentence explaining that.

File: pennylane/__init__.py [hunk 3]
```diff
@@ -78,6 +78,9 @@
     simplify,
 )
 from pennylane.ops.functions import *
+from pennylane.ops import convert
+from pennylane.ops.convert import import_operator
```
Comment (soranjh) [inline pennylane/__init__.py:82]: The module `convert.py` which hosts functions for converting openfermion operator objects to pennylane objects is moved from `qml.qchem` to `qml.ops` because of the generality of the module.

File: tests/ops/test_convert.py [hunk 4]
```diff
@@ -19,12 +19,32 @@
...
-
 import pennylane as qml
-from openfermion import QubitOperator
 from pennylane import numpy as np
-from pennylane import qchem
+
+openfermion = pytest.importorskip("openfermion")
+
+
+@pytest.fixture(
```
Comment (soranjh) [inline tests/ops/test_convert.py:28]: Moved from `qml.qchem.tests.conftest` to here.

File: pennylane/ops/convert.py [hunk 5]
```diff
@@ -281,9 +281,11 @@ def _openfermion_pennylane_equivalent(
     return openfermion_qubit_operator == _pennylane_to_openfermion(coeffs, ops, wires=wires)
 
 
-def import_operator(qubit_observable, format, wires=None, tol=1e08):
+def import_operator(qubit_observable, format="openfermion", wires=None, tol=1e08):
     r"""Converts an external operator to a Pennylane operator.
 
+    The currently supported external format is openfermion.
```
Comment (Jaybsoni) [inline pennylane/ops/convert.py:287]:
```suggestion
    The external format currently supported is openfermion.
```

File: qchem/pennylane_qchem/qchem/obs.py [hunk 6]
```diff
@@ -14,11 +14,12 @@
 """This module contains functions to construct many-body observables whose expectation
 values can be used to simulate molecular properties.
 """
+import numpy as np
+
```
Comment (Jaybsoni) [inline qchem/pennylane_qchem/qchem/obs.py:18]: Why was this moved to the top ? was there a pylint issue or something ?

File: qchem/tests/test_convert_observable.py [hunk 7]
```diff
@@ -1,556 +0,0 @@
-import os
```
Comment (Jaybsoni) [inline qchem/tests/test_convert_observable.py:1]: Out of curiosity, were these tests moved else where, or was the underlying code deprecated?

**Review by Jaybsoni — APPROVED** [review]
Looks good

File: qchem/tests/test_convert_observable.py (hunk 7 above)
Comment (soranjh) [reply qchem/tests/test_convert_observable.py:1]: They were moved to `test_convert.py` in a previous PR.

File: qchem/pennylane_qchem/qchem/obs.py (hunk 6 above)
Comment (soranjh) [reply qchem/pennylane_qchem/qchem/obs.py:18]: I run `isort` to sort the imports.

**Review by obliviateandsurrender — APPROVED** [review]
Looks good to go! Just please check the failing codecov test.
